src/ripple/ledger/impl/ApplyStateTable.cpp

```diff
@@ -520,12 +522,26 @@ ApplyStateTable::destroyXRP(XRPAmount const& fee)
 
 // Insert this transaction to the SLE's threading list
 void
-ApplyStateTable::threadItem(TxMeta& meta, std::shared_ptr<SLE> const& sle)
+ApplyStateTable::threadItem(
+    TxMeta& meta,
+    std::shared_ptr<SLE> const& sle,
+    Rules const& rules)
 {
     key_type prevTxID;
     LedgerIndex prevLgrID;
 
-    if (!sle->thread(meta.getTxID(), meta.getLgrSeq(), prevTxID, prevLgrID))
+    static std::set<LedgerEntryType> newPreviousTxnIDTypes = {
```

Comment:
Please `static const` here. Mutable `static` (that's not actually mutated) in function scope would be an annoying distraction, should we ever need to investigate this part.

Reply:
Good catch, fixed in 93e8884